docs/monthly-meeting/2024-01.md

```diff
@@ -0,0 +1,104 @@
+  - Petr will bounce ideas off Daniele :)
+
+
+- Ryan tried introducing argument lists (like in [sqlite3](https://docs.python.org/3/library/sqlite3.html#sqlite3.connect)) to the collections module, didn't work too well as the functions take one or two args and they're pretty obvious. Petr recommended looking at `subprocess.Popen`.
+  - [Carol] Ryan, feel free to ping me when you have an example since I'm very familiar with the scientific Python docs.
+  - Guido doesn't like forcing people to stick to a template
+  - Daniele mentions there are two kinds of users, one of which -- a forensic analyst -- could need very strictly formatted documentation
+  - [Guido] The bulleted list makes it less likely to fit everything on a screen. Vertical space is a precious resource.
+  - [Carol] Doc usage has changed thanks to intellisense, nowadays you get lists of arguments on hovering.
+  - [Carol] Areas where users ask for help most should get most attention
+  - [Ezio] There is some value in a consistent format for return values and exceptions; sometimes it's hard to find them currently. Arguments are usually explained well in the description.
+  - [Daniele] It doesn't need to be machine-readable. It should always be maximally human-comfortable.
+  - [Jim] It sounds like the framing of this discussion is, source text targeting a big module documentation page with docs for all that module's methods. There are other possible targets: docstrings, tooltips text. Are there other targets for doc source which we should consider in this discussion?
+  - [Ned] We write docs twice -- once for the docstrings and once in `rst`. That's more work, but it means we can have different strategies for each use case. [Guido] Changing that would involve several PEPs.
+  - [CAM] Sphinx has ways to generate docs for a whole module, or just a single function, from docstrings. But yes, starting to use that would need a PEP.
+  - [CAM] The strategy we took for `sqlite3` was to focus on the functions with most arguments, where the bulleted lists have most benefit, and leave the smaller functions for later, when we have more experience with the approach.
+  - [Carol] And we should keep in mind that our perspectives are not necessarily the users' perspective. If we do make major changes we should think about users and we should vet the changes.
+  - Ryan started contributing by trying to get the contract of [`collections.Counter.most_common`](https://docs.python.org/3/library/collections.html#collections.Counter.most_common), and not finding it in the docstring nor in the documentation.
+  - Carol: If you're coming from the scientific world, where the `?` in IPython gives you the docstring, that is often the first thing you look at.
+  - Ryan will take a look at only documenting return type and "raises" in a structured way, leaving parameters in the prose.
+  - Guido warns that `collections.Counter` is not that typical.
```

Comment:
```suggestion
- Ryan tried introducing argument lists (like in [sqlite3](https://docs.python.org/3/library/sqlite3.html#sqlite3.connect)) to the collections module, didn't work too well as the functions take one or two arguments and they're pretty obvious. Petr recommended looking at `subprocess.Popen`.
  - [Carol] Ryan, feel free to ping me when you have an example since I'm very familiar with the scientific Python docs.
  - Guido doesn't like forcing people to stick to a template
  - Daniele mentions there are two kinds of users, one of which -- a forensic analyst -- could need very strictly formatted documentation
  - [Guido] The bulleted list makes it less likely to fit everything on a screen. Vertical space is a precious resource.
  - [Carol] Doc usage has changed thanks to intellisense, nowadays you get lists of arguments on hovering.
  - [Carol] Areas where users ask for help most should get most attention.
  - [Ezio] There is some value in a consistent format for return values and exceptions; sometimes it's hard to find them currently. Arguments are usually explained well in the description.
  - [Daniele] It doesn't need to be machine-readable. It should always be maximally human-comfortable.
  - [Jim] It sounds like the framing of this discussion is, source text targeting a big module documentation page with docs for all that module's methods. There are other possible targets: docstrings, tooltips text. Are there other targets for doc source which we should consider in this discussion?
  - [Ned] We write docs twice -- once for the docstrings and once in `rst`. That's more work, but it means we can have different strategies for each use case.
    - [Guido] Changing that would involve several PEPs.
  - [CAM] Sphinx has ways to generate docs for a whole module, or just a single function, from docstrings. But yes, starting to use that would need a PEP.
  - [CAM] The strategy we took for `sqlite3` was to focus on the functions with most arguments, where the bulleted lists have most benefit, and leave the smaller functions for later, when we have more experience with the approach.
  - [Carol] And we should keep in mind that our perspectives are not necessarily the users' perspective. If we do make major changes we should think about users and we should vet the changes.
  - Ryan started contributing by trying to get the contract of [`collections.Counter.most_common`](https://docs.python.org/3/library/collections.html#collections.Counter.most_common), and not finding it in the docstring nor in the documentation.
  - [Carol] If you're coming from the scientific world, where the `?` in IPython gives you the docstring, that is often the first thing you look at.
  - Ryan will take a look at only documenting return type and "raises" in a structured way, leaving parameters in the prose.
  - Guido warns that `collections.Counter` is not that typical.
```